torch/nn/modules/module.py

```diff
@@ -1164,4 +1166,20 @@ def _replicate_for_data_parallel(self):
         replica._parameters = replica._parameters.copy()
         replica._buffers = replica._buffers.copy()
         replica._modules = replica._modules.copy()
+
+        # Warn users that gradients don't behave as expected on replica modules
+        old_zero_grad = replica.__class__.zero_grad
+        weak_self = weakref.ref(replica)
+
+        def zero_grad():
+            warnings.warn(
+                "The parameters in data parallel modules are copied from the original module."
```

Comment:
nit: Just mention where the warning comes from: "Calling `.zero_grad()` from a module that was passed to a nn.DataParallel() has no effect. The parameters in data parallel modules are copied (in a differentiable manner) from the original module."